common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -138,21 +155,41 @@ public void invoke(@NonNull UsbYubiKeyDevice device) {
                 synchronized (sDeviceLock) {
                     //set device
                     mDevice = device;
+                    //Reset DialogHolder to null if necessary.
+                    //In this case, DialogHolder would be an ErrorDialog if not null.
+                    mDialogHolder.dismissDialog();
+                    //Make sure instance variables are reset/set properly and requests are canceled.
                     mDevice.setOnClosed(new Runnable() {
                         @Override
                         public void run() {
                             synchronized (sDeviceLock) {
                                 Logger.verbose(TAG, "A YubiKey device was disconnected");
                                 mDevice = null;
-                                //Call onCancel on current dialog, if dialog is showing.
-                                synchronized (sSmartcardDialogLock) {
-                                    if (mCurrentDialog != null) {
-                                        mCurrentDialog.onCancelCba();
-                                        //Reset dialog to null
-                                        mCurrentDialog = null;
-                                        //TODO: maybe show a dialog that tells user they unplugged too early? I'm not sure this is necessary since I'm thinking in a lot of cases, the user wanted to cancel current flow by unplugging.
-                                        Logger.verbose(TAG, "YubiKey was disconnected while dialog was still displayed.");
-                                    }
+                                //Remove the YKPiv security provider if it was added.
+                                if (Security.getProvider(YUBIKEY_PROVIDER) != null) {
+                                    //Remove provider.
+                                    Security.removeProvider(YUBIKEY_PROVIDER);
+                                    //Telemet and log that PivProvider is being removed from Security static list.
+                                    Telemetry.emit((BaseEvent) new BaseEvent().put(
+                                            TelemetryEventStrings.Key.PIVPROVIDER_REMOVED,
+                                            String.valueOf(true)
```

Comment:
We can probably expose a method that allows to emit booleans directly...we don't have to do this right now